### PR TITLE
added support for deleting nagios downtime

### PR DESCRIPTION
**Pull request by jonathanbouvier**
##### ISSUE TYPE

- Feature Pull Request
##### COMPONENT NAME

monitoring/nagios.py
##### ANSIBLE VERSION

```
ansible 2.1.0.0
```
##### SUMMARY

This change adds support for removing downtime using the DEL_DOWNTIME_BY_HOST_NAME external command introduced in Nagios Core 3.x: 
https://sourceforge.net/u/c8656/nagios/ci/9f0e69b30d4a9de8c057a304e60c9410937b16ed/tree/include/common.h
https://sourceforge.net/p/nagios/mailman/message/30453206/

I added this functionality because I was writing playbooks to perform long running tasks where the runtimes were highly variable. I wanted to be able to cancel downtime as soon as the task completed, rather than potentially waiting an hour plus for the downtime to expire. I have tested this against Nagios XI R2014 (Nagios Core 4.x), and this external command works flawlessly.

The syntax for the external command is:
DEL_DOWNTIME_BY_HOST_NAME;<host_name>;[<service_desription>];[<start_time>];[<comment>]

It allows one to delete all downtime for a particular host (including services), or limit the removal to a particular combination or service, comment, and start time. I do not think it is possible to ONLY remove host downtime (and not service downtime for the same host) with this command.

I tried to follow the conventions that were already set for services=(host,all,etc).
Setting services=host will submit a delete command with <service_description> blank and <comment> equal to the default (Scheduling downtime), or a provided value.
Setting services=all will submit a delete command with <service_description> and <comment> blank, which has the effect of removing all downtime for the given host.
I didn't see much value in providing a <start_time> flag so this has not been implemented.

### CODE REVIEW

**Comment by gregdek** [comment]
Thanks @jonathanbouvier. To the current maintainers, @tbielawa please review according to guidelines (http://docs.ansible.com/ansible/developing_modules.html#module-checklist) and comment with text 'shipit', 'needs_revision' or 'close_me' as appropriate.

[This message brought to you by your friendly Ansibull-bot.]

File: monitoring/nagios.py [hunk 1]
```diff
@@ -383,6 +396,48 @@ def _fmt_dt_str(self, cmd, host, duration, author=None,
 
         return dt_str
 
+    def _fmt_dt_del_str(self, cmd, host, svc=None, start=None, comment=None):
+        """
+        Format an external-command downtime string.
```
Comment (timlnx) [inline monitoring/nagios.py]: _nit pick_ that should say

> Format an external-command downtime deletion string.

As it is different from `_fmt_dt_str()` which uses the same docstring. Not a blocker.

**Comment by timlnx** [comment]
LGTM :+1:

**Comment by gregdek** [comment]
Thanks again to @jonathanbouvier for this PR, and thanks @tbielawa for reviewing. Marking for inclusion.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by resmo** [comment]
@jonathanbouvier please add a note in the argument description that delete_downtime was added in version 2.2 (as it was done for servicegroup in 2.0).

**Comment by jonathanbouvier** [comment]
I just added in the version 2.2 note, and updated the doc string for _fmt_dt_del_str while I was in there.

**Comment by resmo** [comment]
thanks!

**Pull request merged**